js/index.js

```diff
@@ -239,6 +239,38 @@ onload = function() {
   Inspector.init();
   Inspector.showPane('pane-info');
 
+  /** Loads a `File` object as a package.json. */
+  async function loadPackageJSON(file) {
+    if (file.type && file.type != 'application/json') return alert('File must have a ".json" extension');
+    if (!file) return alert('Please drop a file, not... well... whatever else it was you dropped');
+
+    const reader = new FileReader();
+
+    const content = await new Promise((resolve, reject) => {
+      reader.onload = () => resolve(reader.result);
+      reader.readAsText(file);
+    });
+
+    const module = new Module(JSON.parse(content));
+    history.pushState({module}, null, `${location.pathname}?upload=${file.name}`);
+    graph(module);
+  }
+
+  // Handle file chooser submit
+  const uploadFileRef = $('#upload_file');
+  Object.assign(uploadFileRef, {
+    onsubmit: async ev => {
+      ev.preventDefault();
+
+      // get file from upload `input`
+      const formData = new FormData(uploadFileRef);
+      const file = formData.get('file');
+      if (!file) return alert('No files found in file upload');
+
+      return loadPackageJSON(file);
+    }
+  });
```

Comment:
Pairing with changes above, just watch for change event on the file input?  (Haven't actually tested this, but I think this should work, no?)
```
$('#upload_file').onchange = ev => {
  ev.preventDefault();
  (... etc.)
}
```

Reply:
Implemented this in the new changes.